src/wazuh_db/wdb_syscollector.c

```diff
@@ -153,20 +159,25 @@ int wdb_netproto_save(wdb_t * wdb, const char * scan_id, const char * iface, int
         item_id,
         replace) < 0) {
 
-        return -1;
+        return OS_INVALID;
     }
 
-    return 0;
+    return OS_SUCCESS;
 }
 
 // Insert IPv4/IPv6 protocol info tuple. Return 0 on success or -1 on error.
 int wdb_netproto_insert(wdb_t * wdb, const char * scan_id, const char * iface, int type, const char * gateway, const char * dhcp, int metric, const char * checksum, const char * item_id, const bool replace) {
-
     sqlite3_stmt *stmt = NULL;
 
+    if (NULL == iface) {
+        if(checksum && 0 != strcmp(SYSCOLLECTOR_LEGACY_CHECKSUM_VALUE, checksum)) {
+            wdbi_remove_by_pk(wdb, WDB_SYSCOLLECTOR_NETPROTO, item_id);
+        }
```

Comment:
```suggestion
        wdbi_remove_by_pk(wdb, WDB_SYSCOLLECTOR_NETPROTO, item_id);
```
the item_id is a sync field, no need to re-identify it, check the call tree. (For legacy code, item_id is null)

Reply:
ditto